Problem when linking due to variable declaration in header

Having the definitions of `energymon_path` and `hsmpmon_path` in a header file (`include/e_smi/e_smi_monitor.h`)  leads to linking problems with GCC 11.2

```
...
[ 50%] Linking C shared library libe_smi64.so
/apps/all/CMake/3.21.1-GCCcore-11.2.0/bin/cmake -E cmake_link_script CMakeFiles/e_smi64.dir/link.txt --verbose=1
/apps/all/GCCcore/11.2.0/bin/cc -fPIC -shared -Wl,-soname,libe_smi64.so.1 -o libe_smi64.so.1.5 CMakeFiles/e_smi64.dir/src/e_smi.c.o CMakeFiles/e_smi64.dir/src/e_smi_monitor.c.o CMakeFiles/e_smi64.dir/src/e_smi_utils.c.o  -lpthread -lrt -lm 
/apps/all/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: CMakeFiles/e_smi64.dir/src/e_smi_monitor.c.o: multiple definition of 'energymon_path'
/apps/all/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: CMakeFiles/e_smi64.dir/src/e_smi.c.o: previous definition here
/apps/all/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: error: CMakeFiles/e_smi64.dir/src/e_smi_monitor.c.o: multiple definition of 'hsmpmon_path'
/apps/all/binutils/2.37-GCCcore-11.2.0/bin/ld.gold: CMakeFiles/e_smi64.dir/src/e_smi.c.o: previous definition here
collect2: error: ld returned 1 exit status
...
```

--- a/include/e_smi/e_smi_monitor.h
+++ b/include/e_smi/e_smi_monitor.h
@@ -102,8 +102,6 @@ typedef enum {
 	MONITOR_TYPE_MAX		//!< Max Monitor Type coordinate
 } monitor_types_t;
 
-char energymon_path[DRVPATHSIZ], hsmpmon_path[DRVPATHSIZ];
-
 int read_energy(monitor_types_t type, uint32_t sensor_id, uint64_t *val);
 int batch_read_energy(monitor_types_t type, uint64_t *pval, uint32_t entries);
 

--- a/src/e_smi.c
+++ b/src/e_smi.c
@@ -91,6 +91,8 @@ static const char *proc_str = "processor";
 static const char *apic_str = "apicid";
 static const char *node_str = "physical id";
 
+extern char energymon_path[DRVPATHSIZ], hsmpmon_path[DRVPATHSIZ];
+
 /*
  * To Calculate maximum possible number of cores and sockets,
  * cpu/present and node/possible entires may return 0-127.

--- a/src/e_smi_monitor.c
+++ b/src/e_smi_monitor.c
@@ -96,6 +96,8 @@ static char *filenames[MONITOR_TYPE_MAX] = {energy_file,
 					    socket_temp_mon_file,
 };
 
+char energymon_path[DRVPATHSIZ], hsmpmon_path[DRVPATHSIZ];
+
 int find_energy(char *devname, char *hwmon_name)
 {
 	DIR *pdir;

